src/pilz_github_ci_runner/hardware_tester.py

```diff
@@ -21,17 +21,24 @@
 import subprocess
 
 
+KEYS = ["ROS_DISTRO", "ROS_REPO", "DOCKER_RUN_OPTS", "APT_PROXY", "CMAKE_ARGS"]
+
+
 class HardwareTester(object):
-    def __init__(self, token, log_dir, cmake_args, docker_opts, apt_proxy, setup_cmd, cleanup_cmd, *args, **kwargs):
+    def __init__(self, token, log_dir, ci_args, setup_cmd, cleanup_cmd, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._token = token
         self._log_dir = log_dir
-        self._docker_opts = docker_opts
-        self._cmake_args = cmake_args
-        self._apt_proxy = apt_proxy
+        self._env = self._create_ci_env(ci_args)
         self._setup_cmd = setup_cmd
         self._cleanup_cmd = cleanup_cmd
 
+    def _create_ci_env(self, ci_args):
+        relevant_env = {k: os.environ.get(k) for k in KEYS if k in os.environ}
```

Comment:
"Is it necessary that the variables are actually os environment variables?"
No, It is not neccessary for them to be actually os env variables it will take:
 - system environment vriables
 - variables defined in front of the command aka ROS_DISTRO=melodic rosrun pilz_github...
 - variables passed as CI_ARGS

priority is: CI_ARGS > in fron of command > system environment

"What is the reasoning behind these lines?"
Added another function and renamed this one to hopefully clarify the idea behind this lines. 

For what {k: os.environ.get(k) for k in KEYS if k in os.environ} does in special lookup dict comprehension in python. 
In short: It loops through the KEYS and creates a new dict which it fills with all the os.env variables having such a key and exist.